projects/components/src/snackbar/snackbar.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { MatSnackBar, MatSnackBarConfig, MatSnackBarRef } from '@angular/material/snack-bar';
+import { Subject } from 'rxjs';
+import { SnackbarComponent, SnackbarMode } from './snackbar.component';
+
+@Injectable({ providedIn: 'root' })
+export class SnackbarService {
+  private readonly snackBarDefaultConfig: MatSnackBarConfig = {
+    horizontalPosition: 'left',
+    verticalPosition: 'bottom',
+    duration: 5000 // 5 seconds in ms
+  };
+
+  private snackbarRef?: MatSnackBarRef<unknown>;
+  private readonly closedObserver$: Subject<unknown> = new Subject();
+
+  public constructor(private readonly snackbar: MatSnackBar) {
+    this.closedObserver$.subscribe(() => {
```

Comment:
So this means we never support multiple concurrent snackbars. Generally the case, but is that the explicit intention? If not, we could keep the ref and subject local to each function call with the same effect (not sure if the underlying lib supports multiple or not anyway, I'm assuming it does).

Reply:
the openFromComponent method in MatSnackbar library supports having one snackbar at a time. It dismisses the currently open one when a new one is opened. I was trying to have that single snackbar usecase in this iteration.

Reply:
By the way, do we want to rename this to NotificationService? @aaron-steinfeld

Reply:
I do like that name better. We definitely don't need to build out support for multiple concurrent right now, I was just wondering if we were explicitly hiding it (but sounds like it never existed).

Reply:
I do think we should only show one snack bar at a time. We can show one message for the overall status instead of the individual errors which we can just on the actual page/modal itself.